src/python_testing/tc_mode_base.py

```diff
@@ -0,0 +1,112 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import logging
+
+from mobly import asserts
+from chip.clusters.Types import NullValue
+
+logger = logging.getLogger(__name__)
+
+
+class TC_MODE_BASE:
+
+    def initialize_tc_base(self, endpoint, attributes, requested_cluster, cluster_objects, *args):
```

Comment:
also needs comments. Why is this a separate method and not `__init__` ?

Reply:
I was not sure of the "best" way to handle this additional class (also having problems with super().__init__ when using TC_MODE_BASE along with MatterBaseTest in the test case). I've replaced this function with the __init__ and reduced the parameters, still not sure if this is the "best" way tho.